networkx/algorithms/shortest_paths/weighted.py

```diff
@@ -618,6 +620,9 @@ def multi_source_dijkstra_path_length(G, sources, cutoff=None, weight="weight"):
     """
     if not sources:
         raise ValueError("sources must not be empty")
+    for s in sources:
+        if s not in G:
+            raise nx.NodeNotFound(f"Node {s} not found in graph")
```

Comment:
For the functions where `sources` is a container of nodes, it might be a slight improvement to do something like:

```suggestion
    if set(sources) - set(G):
        raise nx.NodeNotFound(f"Node(s) {set(sources) - set(G)} not found in graph")
```

The reason being, if the user supplies a set of nodes where *more than 1* node isn't in `G`, then this will catch all of the instances rather than iterating through and catching the non-existent nodes one at a time.

This is a very minor consideration and not a blocker here - happy to leave it for a follow-on PR. Just some food for thought!